tests/test_adapters/test_proxy.py

```
@@ -264,6 +265,42 @@ def test_model_injected(self, proxy):
         _, body = _http_post(f"{proxy.url}/v1/chat/completions", {"messages": []})
         assert body["echo_body"]["model"] == "test-model"
 
+    def test_async_stop_closes_endpoint_on_proxy_loop(self, echo_upstream, monkeypatch):
+        from nemo_evaluator.adapters.interceptors import endpoint as endpoint_mod
+
+        original_close = endpoint_mod.Interceptor.close
+        close_calls: list[tuple[int, int, int | None]] = []
+
+        async def close_spy(self):
```

Ruff (0.15.14)

[warning] Missing return type annotation for private function `close_spy`

Add return type annotation: `None`

(ANN202)